gobblin-compaction/src/main/java/gobblin/compaction/HdfsIO.java

```diff
@@ -84,11 +86,15 @@ protected static Configuration getConfiguration() throws FileNotFoundException {
   }
 
   private static void addResourceToConf(Configuration conf) {
-    conf.setStrings(getHdfsUriHadoopPropertyName(), HDFS_URI_DEFAULT);
     addHadoopConfigPropertiesToConf(conf);
+    String hdfsUriKey = getHdfsUriHadoopPropertyName();
     if (CompactionRunner.properties.containsKey(HDFS_URI)) {
-      conf.setStrings(getHdfsUriHadoopPropertyName(), CompactionRunner.properties.getProperty(HDFS_URI));
+      conf.set(hdfsUriKey, CompactionRunner.properties.getProperty(HDFS_URI));
     }
+    if (Strings.isNullOrEmpty(conf.get(hdfsUriKey))) {
```

Comment:
hdfsUriKey can't be null or empty here (and if it is, then the next line doesn't make sense).
I think you can replace the `if` block in line 91-93 with `conf.set(hdfsUriKey, CompactionRunner.properties.getProperty(HDFS_URI, HDFS_URI_DEFAULT));`